src/peer.ts

```diff
@@ -0,0 +1,60 @@
+import net from "net";
+import {EventEmitter} from "events";
+
+export interface PeerOpts {
+  ip: string;
+  port: number;
+}
+
+export default class Peer extends EventEmitter {
+  public ip: string;
+  public port?: number;
+  public connection: net.Socket;
+
+  /**
+   * Constructor
+   * @param {PeerOpts} opts
+   */
+  public constructor(opts: PeerOpts) {
+    super();
+    this.ip = opts.ip;
+    this.port = opts.port || 9000;
+  }
+
+  /**
+   * Establishes a connection to the peer.
+   * @returns {boolean}
+   */
+  public connect = (): Promise<boolean> => {
+    // Attempt to connect to peer, if connection refused remove the peer from bootnodes.
+    const that = this;
+    return new Promise((resolve, reject) => {
+      that.connection = net.createConnection({port: this.port});
+      that.connection.on('connect', resolve);
+      that.connection.on('error', reject);
+    })
+  };
+
+  /**
+   * Starts listening for incoming messages from the server
+   */
+  public start = () => {
+    this.connection.on('data', (data) => {
+      this.emit("new-data", data.toString());
+      this.connection.end();
```

Comment:
I don't think we want to end() here.

We probably want do something like:
```typescript
this.connection.on.('data', (data) => this.processData(data))
```

and
```typescript
public processData(data) {
  switch (dataType(data)) {
    case DATA_HELLO:
      this.emit('hello', parseHelloData(data))
    ...
}
```
Or something to that effect to split out the event types being emitted.

Reply:
Event names should probably be exported enum so we can use this in other parts of code